### PR TITLE
build and push cleaner image to docker hub

### PR DESCRIPTION
**Pull request by jsanda**

### CODE REVIEW

**Comment by jsanda** [comment]
@burmanm I decided to name the image cleaner instead of k8ssandra-client for a couple reasons. First, it only supports the deletion functionality right now. Secondly, I worry that people might be confused by k8ssandra-client in terms of what it does.

File: .github/workflows/main.yaml [hunk 1]
```diff
@@ -133,3 +133,16 @@ jobs:
...
+      - setup
+    runs-on: ubuntu-latest
+    steps:
+    - name: Login to Docker Hub
+      if: github.ref == 'refs/heads/master'
+      run: echo "${{ secrets.DOCKER_HUB_PASSWORD }}" | docker login docker.io --username "${{ secrets.DOCKER_HUB_USERNAME }}" --password-stdin
+
+    - name: Build and push image
+      if: github.ref == 'refs/heads/master'
+      run: make cleaner-docker-build cleaner-docker-push
```
Comment (bradfordcp) [inline .github/workflows/main.yaml:148]: https://i.imgflip.com/zhsy3.jpg

File: .github/workflows/main.yaml [hunk 2]
```diff
@@ -133,3 +133,16 @@ jobs:
         run: |
           export PATH=$GOROOT/bin:$GOPATH/bin:$PATH
           make test
+
+  build-image:
+    needs:
+      - setup
+    runs-on: ubuntu-latest
```
Comment (burmanm) [inline .github/workflows/main.yaml:140]: Should this be another workflow? And only run it if pkg/cleaner was modified or cmd/cleaner ..?

Should be doable by setting it in the on<pull_request>.paths

**Comment by jsanda** [comment]
@burmanm can you take another look at this? I added a new workflow.

**Comment by burmanm** [comment]
You forgot to push?

**Comment by jsanda** [comment]
@burmanm I had pushed to my fork 🤦  I just pushed to origin.

File: .github/workflows/build-cleaner.yaml [hunk 3]
```diff
@@ -0,0 +1,23 @@
+---
+on:
+  pull_request:
```
Comment (burmanm) [inline .github/workflows/build-cleaner.yaml:3]: On push or pull_request? I would assume push to the main

Comment (jsanda) [reply .github/workflows/build-cleaner.yaml:3]: Yeah, good point!

**Review by burmanm — APPROVED** [review]

**Pull request merged**